packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3463,6 +3469,30 @@ export class ContainerRuntime
 		return result;
 	}
 
+	detachHead(): { merge(): void; pause(): Promise<void>; dispose(): void } {
+		const checkpoint = this.outbox.getBatchCheckpoints(true);
+		let paused = false;
+		const branchInfo = {
+			dispose: () => {
+				checkpoint.mainBatch.rollback();
```

Comment:
we should assert the other batches are empty for now. we'll have to figure out rollback for those. id compressor probably doesn't matter, but we'll need to actually undo attaches.